### PR TITLE
Add support for the Strand-specific coordinate count datatype

### PR DESCRIPTION
**Pull request by gregvonkuster**
(extension scidx) which is used by tools within the soon-coming
Chip-exo Galaxy flavor (see
https://github.com/galaxyproject/tools-iuc/issues/404).  The Chip-exo
Galaxy flavor will initially be used by the Center for Eukaryotic Gene
Regulation labs at Penn State University.

### CODE REVIEW

**Comment by martenson** [comment]
@galaxybot test this

File: lib/galaxy/datatypes/interval.py [hunk 1]
```diff
@@ -1482,6 +1482,86 @@ def sniff( self, filename ):
...
+        # Don't set column names since the first
+        # line of the dataset displays them.
+        self.column_names = []
+
+    def sniff(self, filename):
+        """
+        Checks for 'scidx-ness.'
+        """
+
+        def is_scidx(fh, bool):
```
Comment (bgruening) [inline lib/galaxy/datatypes/interval.py]: @gregvonkuster can you use try, (except), finally instead? https://docs.python.org/2.5/whatsnew/pep-341.html

File: lib/galaxy/datatypes/interval.py [hunk 2]
```diff
@@ -1482,6 +1482,86 @@ def sniff( self, filename ):
...
+                        return is_scidx(fh, False)
+                    value = items[4]
+                    if not value.isdigit():
+                        return is_scidx(fh, False)
+                    if int(forward) + int(reverse) != int(value):
+                        return is_scidx(fh, False)
+                if count == 100:
+                    return is_scidx(fh, True)
+                count += 1
+            if count < 100 and count > 0:
```
Comment (bgruening) [inline lib/galaxy/datatypes/interval.py:1553]: Is the indentation here correct? You aim was probably to abort return after 100 successful lines true, right?

Comment (gregvonkuster) [inline lib/galaxy/datatypes/interval.py:1553]: Actually, this case covers the scenario where the files is less than 100 lines, but has not yet failed any checks, so this check is outside of the while.  The scenario where the files is longer than 100 lines, but has not failed any checks is handled by the "if count == 100:" check 3 lines above.

File: lib/galaxy/datatypes/interval.py (hunk 1 above)
Comment (gregvonkuster) [inline lib/galaxy/datatypes/interval.py]: Ok, working on it. Cleaned up a bit - thanks @bgruening  ;)

**Comment by martenson** [comment]
@galaxybot test this

**Comment by martenson** [comment]
```
./lib/galaxy/datatypes/interval.py:1534:37: E225 missing whitespace around operator
./lib/galaxy/datatypes/interval.py:1555:27: F841 local variable 'e' is assigned to but never used
```

**Comment by gregvonkuster** [comment]
@martenson Sorry - I usually flake8 stuff, but I missed this one.  Fixes for these are in my latest.

**Comment by martenson** [comment]
@galaxybot test this

**Comment by gregvonkuster** [comment]
ping @martenson @bgruening I have a PR ready that will handle this https://github.com/galaxyproject/galaxy/issues/667#issuecomment-159305155, but I'm waiting to get this one merged.  Can this be merged, or is there more I need to do?  Thanks!

**Comment by martenson** [comment]
Something weird happened to this PR (See commits), I added WIP label and please do not merge until it is resolved. See email for details.

**Comment by gregvonkuster** [comment]
@martenson @bgruening I'll just come it an resubmit it.  Thanks.

**Comment by gregvonkuster** [comment]
Closing due to some strange corruption.

**Pull request closed without merging**